src/core/src/pass/constant_folding.cpp

```diff
@@ -120,9 +129,15 @@ bool ov::pass::ConstantFolding::pre_calculated_values_folding(const std::shared_
             // propagation because we can't detect borders of shape_of sub-graphs, so we propagate can_be_folded
             // attribute through all nodes including nodes on data path. So to limit the spread of attribute to other
             // shape-of sub-graphs we do not propagate it through ShapeOf nodes.
-            can_be_folded = true;
+            can_be_folded = input_values.begin()->get_partial_shape().is_static();
+        } else if (op::util::is_parameter(node) || op::util::is_output(node) || op::util::is_sink(node) ||
+                   is_type<ov::opset3::ReadValue>(node) || is_type<ov::opset6::ReadValue>(node)) {
```

Comment:
The same here... Because we introduce more and more `if`s for some specific operations

Reply:
> The same here... Because we introduce more and more `if`s for some specific operations

Parameters, Result, Read/Assign values are not regular operation and they have a special meaning. We can't just ignore it and handle them like others.

Reply:
please use ReadValueBase class